### PR TITLE
aiohttp instrumentation: correct url filter input type

### PR DESCRIPTION
**Pull request by Olegt0rr**
# Description

Url filter function receives `yarl.URL` type, as described in the documentation.
But `_UrlFilterT` type describes `str` input... We need to fix it

Screenshot 2022-01-15 at 02 57 07

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def response_hook(span: Span, params: typing.Union[
 from opentelemetry.trace.status import Status, StatusCode
 from opentelemetry.util.http import remove_url_credentials
 
-_UrlFilterT = typing.Optional[typing.Callable[[str], str]]
+_UrlFilterT = typing.Optional[typing.Callable[[yarl.URL], str]]
```
Comment (owais) [inline instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py:111]: Did any older version use `str`? In that case probably better to have a union type.

Comment (Olegt0rr) [reply instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py:111]: No, it's `yarl.URL` more than 4 years
https://github.com/aio-libs/aiohttp/blame/0e79469381418107e81670dc4e299ad56fdc5354/aiohttp/tracing.py#L214

**Review by lzchen — APPROVED** [review]

**Review by owais — APPROVED** [review]

**Comment by owais** [comment]
@Olegt0rr please resolve conflicts and make sure CI checks pass.

**Review by owais — APPROVED** [review]

**Comment by Olegt0rr** [comment]
@owais,
Please, help.
I can't undestand how to fix docs error `reference target not found: yarl.URL`

**Comment by owais** [comment]
Look at the docs tox job in tox.ini. My guess is that you might have to add yarl as an explicit dependency to that job.

File: docs-requirements.txt [hunk 2]
```diff
@@ -37,3 +37,4 @@ sqlalchemy>=1.0
 tornado>=5.1.1
 ddtrace>=0.34.0
 httpx~=0.18.0
+yarl==1.7.2
```
Comment (ocelotl) [inline docs-requirements.txt:40]:
```suggestion
```

**Review by ocelotl — APPROVED** [review]

**Pull request merged**